keyboards/adpenrose/kintsugi/keymaps/franky/keymap.c

```diff
@@ -0,0 +1,173 @@
+/* Copyright 2021 adpenrose
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include <stdio.h>
+
+/* Constants: */
+char wpm_str[4]; /* Used in the wpm counter function. */
+uint8_t selected_layer = 0; /* Used to change the layer using the encoder. */
```

Comment:
Neither should be needed.

```suggestion
```

Reply:
They are needed for the WPM counting function (wpm_str) and the capability of switching layers using the encoder (selected_layer).

Reply:
only if you use s(n)printf.